### ISSUE TITLE
Is the gradient penalty loss problematic when input image is large?

### ISSUE THREAD

**Issue by shaform** [kodalinaveen3/DRAGAN#4]
Hi,
Here the sum of squares is computed
`slopes = tf.sqrt(tf.reduce_sum(tf.square(gradients), reduction_indices=[1]))`

However, when the input image is extremely large, the dimension of gradients would be huge. It seems possible that the resulting slopes would be extremely large compared to 1.
Is this the case?

**Comment by kodalinaveen3** [comment]
We take gradients of D(x) in the x-space. Because we only perturb samples slightly, the size of input image shouldn't matter that much
